### PR TITLE
Fix issue with cross-dep bins not symlinking

### LINKED ISSUE #126: Binary exported by workspace package causes "bolt install" failure
| Title               | Description
| --------------------|------------
| Version             | 0.19.2
| Type                | Issue
| node                | 8.9.4
| Operating System    | macOS 10.13.2
| Short Description   | Binary exported by workspace package causes "bolt install" failure
| Detailed description| If I have a bolt workspace containing a package that has a `"bin"` entry in its `package.json` file, and also have another package in the same workspace that depends on the first bin-exporting package, then running `bolt` results in an error.

The error displayed is:

```sh
$ bolt
⚡️   bolt v0.19.2 (node v8.9.4)
🔎   [1/3] Validating project...
📦   [2/3] Installing project dependencies...
yarn install v1.3.2
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
✨  Done in 0.14s.
🔗   [3/3] Linking workspace dependencies...
error my-bin is linked to a location outside of project node_modules: ../../packages/my-bin/index.js
```

Demo repo here: https://github.com/bengummer/bolt-binary-issue-demo

`yarn install` works in the demo repo, `bolt install` doesn't

### PR DESCRIPTION
**Pull request by jamiebuilds**
Fixes #126

### CODE REVIEW

File: src/commands/install.js [hunk 1]
```diff
@@ -63,7 +63,7 @@ export async function install(opts: InstallOptions) {
   });
 
   for (let workspace of workspaces) {
-    const dependencies = workspace.pkg.getAllDependencies().keys();
+    let dependencies = Array.from(workspace.pkg.getAllDependencies().keys());
```
Comment (jamiebuilds) [inline src/commands/install.js]: This was basically the whole thing

Comment (lukebatchelor) [reply src/commands/install.js]: Do you recall what the Array.from was for here @jamiebuilds ?

**Comment by lukebatchelor** [comment]
Moved to #191

**Comment by lukebatchelor** [comment]
I'll leave this branch here juuust in case I've messed something up in the rebase ;)

**Pull request closed without merging**